zerver/lib/export.py

```diff
@@ -2522,3 +2533,42 @@ def get_realm_exports_serialized(user: UserProfile) -> list[dict[str, Any]]:
             pending=pending,
         )
     return sorted(exports_dict.values(), key=lambda export_dict: export_dict["id"])
+
+
+def get_migrations_by_app() -> AppMigrations:
+    migration_status = get_migration_status()
+    migrations_by_app: AppMigrations = {}
+    current_app = None
+
+    for line in migration_status.splitlines():
+        line = line.strip()
+        no_migrations = "(no migrations)"
+
+        if line and not line.startswith("[") and no_migrations not in line:
+            current_app = line
+            migrations_by_app.setdefault(current_app, [])
+        elif current_app and line.startswith("["):
+            migration = line
+            migrations_by_app[current_app].append(migration)
+        elif current_app and no_migrations in line:
+            migrations_by_app[current_app].append(no_migrations)
```

Comment:
This is not combined with the `elif current_app and line.startswith("["):` because the line with no migrations has extra formatting on it.